### PR TITLE
Added Permalink Feature

### PR DESCRIPTION
**Pull request by Maltysen**
A major component missing from the web interpreter was a permalink feature. I have added that in now and it captures the code, input, and whether or not the debug checkbox was checked. It uses get url parameters and uses Jquery's url parameter encoding function. I have also implemented a few other things including a key binding in the text-boxes such that ctrl-enter is a shortcut to run the code and made some slight formatting changes to accommodate the debug window and made the output scrollable when it gets too large. Last, I also made the default for the debug checkbox checked.

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -6,12 +6,12 @@
 
 app = Flask(__name__, template_folder='.', static_folder='.')
 
-@app.route('/')
+@app.route('/', methods=['GET'])
```
Comment (Tyilo) [inline server.py]: `methods=['GET']` is the default, so it's not needed here

File: server.py [hunk 2]
```diff
@@ -45,4 +45,4 @@ def other(path):
 	return app.send_static_file(path)
 
 if __name__ == '__main__':
-	app.run(debug=True)
+	app.run(debug=True, host='0.0.0.0')
```
Comment (Tyilo) [inline server.py]: You probably don't want to have a debug version of this accessible from outside your computer.

File: index.html [hunk 3]
```diff
@@ -100,17 +107,25 @@
...
 		<script>
+			function permalink() {
+				params={code: $('#code').val()}
+				if ($('#input').val()) params.input=$('#input').val()
+				if (!$('#debug').is(':checked')) params.debug=0
+				document.location="?"+$.param(params)
+			}
+
+			function submit_code() {
+                            $.post('/submit', {
```
Comment (Tyilo) [inline index.html]: There's something wrong with the indentation here.

**Comment by Maltysen** [comment]
Oh lol, forgot to change back the debug things. And i don't know why I have it set to get. I'll change it..

**Comment by Maltysen** [comment]
There, fixed.

**Comment by isaacg1** [comment]
It looks solid, and the permalink stuff is a big improvement. I'm going to merge now.

**Comment by isaacg1** [comment]
Damn it, I misclicked on close when I wanted to merge.

**Pull request merged**